alumet/tests/late_source_creation.rs

```diff
@@ -0,0 +1,103 @@
+use anyhow::Context;
+use tokio::{runtime::Runtime, time::timeout};
+use std::{thread, time::{self, Duration}};
+
+pub struct Metrics {
+    pub time_tot: TypedMetricId<u64>,
+}
+
+pub struct MyTestPluginLateMetricCreation {
+    metrics: Option<Metrics>,
+}
+
+#[derive(Debug)]
+struct MyTestSourcePlugin {
+    value: TypedMetricId<u64>,
+}
+
+
+impl AlumetPlugin for MyTestPluginLateMetricCreation {
+    fn name() -> &'static str {
+        "late_source_creation"
+    }
+
+    fn version() -> &'static str {
+        env!("CARGO_PKG_VERSION")
+    }
+
+    fn default_config() -> anyhow::Result<Option<ConfigTable>> {
+        let config = serialize_config(Duration::from_secs(1))?;
+        Ok(Some(config))
+    }
+
+    fn init(_config: ConfigTable) -> anyhow::Result<Box<Self>> {
+        Ok(Box::new(MyTestPluginLateMetricCreation {
+            metrics: None,
+        }))
+    }
+
+    fn start(&mut self, alumet: &mut AlumetPluginStart) -> anyhow::Result<()> {
+        let usec: PrefixedUnit = PrefixedUnit::micro(Unit::Second);
+        let usec_metric = alumet.create_metric("A",usec, "A random metric to test late metric creation inside post_pipeline_start")?;
+        self.metrics = Some(Metrics {
+            time_tot: usec_metric,
+        });
+        // No source creation in the start function
+        Ok(())
+    }
+
+    fn stop(&mut self) -> anyhow::Result<()> {
+        Ok(())
+    }
+
+    fn post_pipeline_start(&mut self, alumet: &mut AlumetPostStart) -> anyhow::Result<()> {
+        let control_handle = alumet.pipeline_control();
+        let probe = MyTestSourcePlugin{
+            value:(self.metrics.as_ref().expect("Can't read byte_metric")).time_tot,
+        };
+
+        // Add the probe to the sources
+        control_handle
+            .add_source(
+                "x",
+                Box::new(probe),
+                TriggerSpec::at_interval(Duration::from_secs(1)),
+            )
+            .with_context(|| format!("failed to add source when testing add source in post_pipeline_start"))?;
+        Ok(())                         
+    }
+}
+
+
+impl alumet::pipeline::Source for MyTestSourcePlugin {
+    fn poll(&mut self, measurements: &mut MeasurementAccumulator, timestamp: Timestamp) -> Result<(), PollError> {
+        Ok(())
+    }
+}
+
+
+#[test]
+fn late_source_creation_test() {
+    /* This function test for a source add in the post_pipeline_start function */
+    let plugins = static_plugins![MyTestPluginLateMetricCreation];
+    
+    let agent = AgentBuilder::new(plugins)
+        .config_value(toml::Table::new())
+        .build();
+
+    // Stop the pipeline
```

Comment:
This comment is a bit weird, lines 91 to 102 do much more than "stop the pipeline"